app/models/post.rb

```diff
@@ -0,0 +1,27 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', counter_cache: :post_counter
```

Comment:
- [ Required ] I think it would be better if you follow the project requirements when updating the `post_counter` field, I know that you can update the `post_counter` field using `counter_cache` but according to the project requirements, this field should be updated using the `update_post_counter` method so let's follow the project guidelines. Please see the comment I've left on the `update_post_counter` method.

## Project requirements
![Screenshot from 2023-07-20 00-36-11](https://github.com/Elhambasir/blog-app/assets/16392046/2807df97-3496-4b61-a476-932131e188e6)